server/mcp_server_vefaas_function/src/mcp_server_vefaas_function/vefaas_server.py

```diff
@@ -541,40 +532,54 @@ def python_zip_implementation(folder_path: str) -> bytes:
     print(f"Python zip finished, size: {buffer.tell() / 1024 / 1024:.2f} MB")
     return buffer.getvalue()
 
-@mcp.tool(description="""Uploads a code folder to tos and returns the TOS object URL.
-          After it is uploaded, ask the user to release the function again for the changes to take effect.
-Use this when you need to upload a code folder to TOS for a veFaaS function.""")
-def upload_to_tos(region: str, folder_path: str, function_id: str) -> bytes:
+@mcp.tool(description="""
+Uploads code to TOS for a veFaaS function deployment. You can either provide a local folder path or a mapping of filenames to in-memory file content (str or bytes).
+
+- For large or structured projects, it's recommended to use the 'folder_path' parameter to upload a full directory from disk.
+- For lightweight projects, such as single-page web apps or small scripts, you can use 'file_dict' to upload in-memory files directly.
+
+After uploading, remind the user to release the function again for the changes to take effect.
+""")
+def upload_code(region: str, function_id: str, folder_path: Optional[str] = None, file_dict: Optional[dict[str, Union[str, bytes]]] = None) -> bytes:
```

Comment:
only work at local mode, for any remote deployment, this method does not work any more